### ISSUE TITLE
[Issue]: Thousands of "Error reading image tag"

### ISSUE THREAD

**Issue by jgoerzen** [jellyfin/jellyfin#8869]
### Please describe your bug

While scanning my photos library, I get thousands of errors like this one:

```
[2022-12-07 18:20:34.978 +00:00] [ERR] [39] Emby.Photos.PhotoProvider: Image Provider - Error reading image tag for "/Pictures/2010s/2011/09/water-splashing/IMG_1592.JPG"
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at TagLib.IFD.IFDReader.StartIFDLoopDetect()
   at TagLib.IFD.IFDReader.Read(Int32 count)
   at TagLib.Jpeg.File.ReadAPP1Segment(UInt16 length)
   at TagLib.Jpeg.File.ReadMetadata()
   at TagLib.Jpeg.File.Read(ReadStyle propertiesStyle)   at TagLib.File.Create(IFileAbstraction abstraction, String mimetype, ReadStyle propertiesStyle)
   at Emby.Photos.PhotoProvider.FetchAsync(Photo item, MetadataRefreshOptions options, CancellationToken cancellationToken)
```

What's more, it appears that these items are never actually indexed, as it does this lengthy scan again and again (taking hours each time).

I would be happy to send a sample image to someone privately.

The image tags, if they exist, would have been written by digikam.

This is Jellyfin 10.8.8

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS: Debian
- Virtualization: Docker (official image)
- Clients: Firefox
- Browser: Firefox
- FFmpeg Version: 
- Playback Method: n/a
- Hardware Acceleration: n/a
- Plugins: default only
- Reverse Proxy: n/a
- Base URL:
- Networking: n/a
- Storage:
```

### Jellyfin logs

```shell
See above.  That message is repeated thousands of times due to the number of photos in my library.
```

### FFmpeg logs

```shell
n/a
```

### Please attach any browser or client logs here

n/a

### Please attach any screenshots here

n/a

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jgoerzen** [comment]
Here are some screenshots.

First, the playlist as it should be:

![playlist](https://user-images.githubusercontent.com/112925/206797490-e61da76c-cedc-4a70-bded-bb867258143d.png)

Now, the queue as it got set:

![queue](https://user-images.githubusercontent.com/112925/206797511-3c254fda-930b-4820-a96a-51b2dcbbebc6.png)

Also, I've never had any shuffle option on and confirmed shuffle was off everywhere.

**Comment by Bond-009** [comment]
Looks like a TagLib issue

**Comment by chicagobuss** [comment]
this issue is affecting me with version 10.8.13 running on windows 11 pro

slightly different error though:

```
[2024-02-27 13:46:02.415 -06:00] [ERR] [14] Emby.Photos.PhotoProvider: Image Provider - Error reading image tag for "F:\Pictures\DSC03069.JPG"
System.NotImplementedException: Unknown type/count 8/0 (128)
   at TagLib.IFD.IFDReader.CreateIFDEntry(UInt16 tag, UInt16 type, UInt32 count, Int64 baseOffset, ByteVector offsetData, UInt32 maxOffset)
   at TagLib.IFD.IFDReader.ReadIFD(Int64 baseOffset, UInt32 offset, UInt32 maxOffset)
   at TagLib.IFD.IFDReader.ParseMakernote(UInt16 tag, UInt16 type, UInt32 count, Int64 baseOffset, UInt32 offset)
   at TagLib.IFD.IFDReader.CreateIFDEntry(UInt16 tag, UInt16 type, UInt32 count, Int64 baseOffset, ByteVector offsetData, UInt32 maxOffset)
   at TagLib.IFD.IFDReader.ReadIFD(Int64 baseOffset, UInt32 offset, UInt32 maxOffset)
   at TagLib.IFD.IFDReader.Read(Int32 count)
   at TagLib.IFD.IFDReader.ParseIFDEntry(UInt16 tag, UInt16 type, UInt32 count, Int64 baseOffset, UInt32 offset)
   at TagLib.IFD.IFDReader.CreateIFDEntry(UInt16 tag, UInt16 type, UInt32 count, Int64 baseOffset, ByteVector offsetData, UInt32 maxOffset)
   at TagLib.IFD.IFDReader.ReadIFD(Int64 baseOffset, UInt32 offset, UInt32 maxOffset)
   at TagLib.IFD.IFDReader.Read(Int32 count)
   at TagLib.Jpeg.File.ReadAPP1Segment(UInt16 length)
   at TagLib.Jpeg.File.ReadMetadata()
   at TagLib.Jpeg.File.Read(ReadStyle propertiesStyle)   at TagLib.File.Create(IFileAbstraction abstraction, String mimetype, ReadStyle propertiesStyle)
   at Emby.Photos.PhotoProvider.FetchAsync(Photo item, MetadataRefreshOptions options, CancellationToken cancellationToken)
```

**Comment by dschrempf** [comment]
I also have this problem. I get the same error as original poster.

**Comment by divStar** [comment]
I am also having this issue with supposed JPEG files (at least they do work as JPEG files on my system).

It seems, that Jellyfin uses the most recent version TagLibSharp (v2.3.0). It in turn uses ImageSharp v1.0.4 (see https://github.com/mono/taglib-sharp/blob/main/src/TaglibSharp.Tests/TaglibSharp.Tests.csproj). Now while I am not sure if it's e.g. a WebP file, before I got here, I found https://stackoverflow.com/questions/70792715/getting-missing-soi-marker-from-jpegdecoderr and https://github.com/SixLabors/ImageSharp/issues/121, which contain some info about missing SOI markers.

I also found, that [ImageSharp](https://github.com/SixLabors/ImageSharp)'s newest version is v3.1.4 (not v1.0.4), meaning, that TagLibSharp is old.

Therefore they should introduce a fix if possible, though seeing as major versions have changed, I'd expect breaking changes. Perhaps someone can make a PR over there and then perhaps over here, but sadly I don't have the time necessary to do this properly.
